nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -312,18 +312,26 @@ def __init__(
         self.setup_streaming_params()
         self.export_cache_support = False
 
+    def update_max_seq_length(self, seq_length: int, device):
+        # Find global max audio length across all nodes
+        if torch.distributed.is_initialized():
+            global_max_len = torch.tensor([seq_length], dtype=torch.float32, device=device)
+
+            # Update across all ranks in the distributed system
+            torch.distributed.all_reduce(global_max_len, op=torch.distributed.ReduceOp.MAX)
+
+            seq_length = global_max_len.int().item()
+
+        if seq_length > self.max_audio_length:
+            self.set_max_audio_length(seq_length)
+
     def set_max_audio_length(self, max_audio_length):
         """
         Sets maximum input length.
         Pre-calculates internal seq_range mask.
         """
         self.max_audio_length = max_audio_length
         device = next(self.parameters()).device
-        seq_range = torch.arange(0, self.max_audio_length, device=device)
-        if hasattr(self, 'seq_range'):
-            self.seq_range = seq_range
-        else:
-            self.register_buffer('seq_range', seq_range, persistent=False)
```

Comment:
Doesn't this break older mkdels / export

Reply:
No, it is fine. I just now create it on the fly instead of slicing from a predefined tensor.